server/src/main/java/org/apache/kafka/server/SimpleApiVersionManager.java

```diff
@@ -0,0 +1,108 @@
+    private final Set<ApiKeys> enabledApis;
+    private final Features<SupportedVersionRange> brokerFeatures;
+    private final boolean enableUnstableLastVersion;
+    private final Supplier<FinalizedFeatures> featuresProvider;
+    private final ApiVersionsResponseData.ApiVersionCollection apiVersions;
+
+    /**
+     * SimpleApiVersionManager constructor
+     * @param listenerType the listener type
+     * @param enabledApis the enabled apis, which are computed by the listener type
+     * @param brokerFeatures the broker features
+     * @param enableUnstableLastVersion whether to enable unstable last version, see
+     *   {@link org.apache.kafka.server.config.ServerConfigs#UNSTABLE_API_VERSIONS_ENABLE_CONFIG}
+     * @param featuresProvider a provider to the finalized features supported
+     */
+    public SimpleApiVersionManager(ApiMessageType.ListenerType listenerType,
+                                   Set<ApiKeys> enabledApis,
+                                   Features<SupportedVersionRange> brokerFeatures,
+                                   boolean enableUnstableLastVersion,
+                                   Supplier<FinalizedFeatures> featuresProvider) {
+        this.listenerType = listenerType;
+        this.enabledApis = enabledApis;
+        this.brokerFeatures = brokerFeatures;
+        this.enableUnstableLastVersion = enableUnstableLastVersion;
+        this.featuresProvider = featuresProvider;
+        this.apiVersions = ApiVersionsResponse.collectApis(listenerType, enabledApis, enableUnstableLastVersion);
+    }
+
+    public SimpleApiVersionManager(ApiMessageType.ListenerType listenerType,
+                                   boolean enableUnstableLastVersion,
+                                   Supplier<FinalizedFeatures> featuresProvider) {
+        this(listenerType,
+             ApiKeys.apisForListener(listenerType),
+             BrokerFeatures.defaultSupportedFeatures(enableUnstableLastVersion),
+             enableUnstableLastVersion,
+             featuresProvider);
+    }
+
+    @Override
+    public boolean enableUnstableLastVersion() {
+        return enableUnstableLastVersion;
+    }
+
+    @Override
+    public ApiMessageType.ListenerType listenerType() {
+        return listenerType;
+    }
+
+    @Override
+    public Set<ApiKeys> enabledApis() {
+        return enabledApis;
+    }
+
+    @Override
+    public ApiVersionsResponse apiVersionResponse(int throttleTimeMs, boolean alterFeatureLevel0) {
+        FinalizedFeatures currentFeatures = features();
+        return new ApiVersionsResponse.Builder()
+                .setThrottleTimeMs(throttleTimeMs)
+                .setApiVersions(apiVersions)
+                .setSupportedFeatures(brokerFeatures)
+                .setFinalizedFeatures(currentFeatures.finalizedFeatures())
+                .setFinalizedFeaturesEpoch(currentFeatures.finalizedFeaturesEpoch())
+                .setZkMigrationEnabled(false)
```

Comment:
ditto, The `zkMigrationEnabled` setting defaults to `false`, and the ZooKeeper migration feature has been removed in version 4.0. Given this, I believe we can remove `setZkMigrationEnabled(false)`